app/src/test/java/com/github/se/travelpouch/model/activity/ActivityRepositoryUnitTest.kt

```diff
@@ -113,16 +119,49 @@ class ActivityRepositoryUnitTest {
 
   @Test
   fun addActivity_shouldCallFirestoreCollection() {
-    `when`(mockDocumentReference.set(any())).thenReturn(Tasks.forResult(null)) // Simulate success
 
-    // This test verifies that when we add a new event, the Firestore `collection()` method is
-    // called.
-    activityRepositoryFirestore.addActivity(activity, onSuccess = {}, onFailure = {})
-
-    shadowOf(Looper.getMainLooper()).idle()
-
-    // Ensure Firestore collection method was called to reference the "events" collection
-    verify(mockDocumentReference).set(any())
+    val mockFirebaseFirestore: FirebaseFirestore = mock()
+    val mockCollectionReference: CollectionReference = mock()
+    val mockActivityDocumentReference: DocumentReference = mock()
+    val mockEventDocumentReference: DocumentReference = mock()
+
+    val mockVoidTask: Task<Void> = mock()
+
+    whenever(mockFirebaseFirestore.collection(anyOrNull())).thenReturn(mockCollectionReference)
+    whenever(mockCollectionReference.document(anyOrNull()))
+        .thenReturn(mockActivityDocumentReference)
+    whenever(mockEventDocumentReference.id).thenReturn("qwertzuiopasdfghjklm")
+
+    whenever(mockFirebaseFirestore.runTransaction<Void>(anyOrNull())).thenReturn(mockVoidTask)
+    whenever(mockVoidTask.addOnSuccessListener(anyOrNull())).thenReturn(mockVoidTask)
+    whenever(mockVoidTask.addOnFailureListener(anyOrNull())).thenReturn(mockVoidTask)
+    whenever(mockVoidTask.isSuccessful).thenReturn(true)
+
+    var succeeded = false
+    var failed = false
+
+    val activityRepositoryFirebase = ActivityRepositoryFirebase(mockFirebaseFirestore)
+    activityRepositoryFirebase.addActivity(
+        activity, { succeeded = true }, { failed = true }, mockEventDocumentReference)
+
+    val onSuccessListenerCaptor = argumentCaptor<OnSuccessListener<Void>>()
+    verify(mockVoidTask).addOnSuccessListener(onSuccessListenerCaptor.capture())
+    onSuccessListenerCaptor.firstValue.onSuccess(mockVoidTask.result)
+
+    assert(succeeded)
+    assertFalse(failed)
+
+    //    `when`(mockDocumentReference.set(any())).thenReturn(Tasks.forResult(null)) // Simulate
+    // success
+    //
+    //    // This test verifies that when we add a new event, the Firestore `collection()` method is
+    //    // called.
+    //    activityRepositoryFirestore.addActivity(activity, onSuccess = {}, onFailure = {})
+    //
+    //    shadowOf(Looper.getMainLooper()).idle()
+    //
+    //    // Ensure Firestore collection method was called to reference the "events" collection
+    //    verify(mockDocumentReference).set(any())
```

Comment:
Please remove these lines if they are not used.

Reply:
Done